### ISSUE TITLE
WebSocket Server connection fails

### ISSUE THREAD

**Issue by rumkin** [bigstepinc/jsonrpc-bidirectional#23]
version: 5.2.5
node: v8.9.4
os: ubuntu 16.04

### Do

Add connection to jsonrpc server:
```javascript
const rpcServer = new JSONRPC.Server();
    const wsRouter = new JSONRPC.BidirectionalWebsocketRouter(rpcServer);

    rpcServer.registerEndpoint(new TestEndpoint('Test', '/api', {}, JSONRPC.Client));
    rpcServer.addPlugin(new JSONRPC.Plugins.Server.AuthenticationSkip());
    rpcServer.addPlugin(new JSONRPC.Plugins.Server.AuthorizeAll());

    const wsServer = new WebSocket.Server({port: 8080});
    wsServer.on('connection', (ws) => {
        const clientID = wsRouter.addWebSocketSync(ws); // line #31 from trace
        // ...
    });
```

### Result
Caught error on creating adding websocket connection to server's router:

```
/node_modules/jsonrpc-bidirectional/src/BidirectionalWebsocketRouter.js:52
		const strEndpointPath = JSONRPC.EndpointBase.normalizePath(webSocket.url ? webSocket.url : webSocket.upgradeReq.url);
		                                                                                                                ^

TypeError: Cannot read property 'url' of undefined
    at BidirectionalWebsocketRouter.addWebSocketSync (/node_modules/jsonrpc-bidirectional/src/BidirectionalWebsocketRouter.js:52:115)
    at WebSocketServer.wsServer.on (/index.js:31:32)
    at emitTwo (events.js:126:13)
    at WebSocketServer.emit (events.js:214:7)
    at handleUpgrade (/node_modules/ws/lib/websocket-server.js:89:16)
    at WebSocketServer.completeUpgrade (/node_modules/ws/lib/websocket-server.js:282:5)
    at WebSocketServer.handleUpgrade (/node_modules/ws/lib/websocket-server.js:221:10)
    at Server.WebSocketServer._ultron.on (/node_modules/ws/lib/websocket-server.js:88:14)
    at emitThree (events.js:136:13)
    at Server.emit (events.js:217:7)
```

This is beacuse ws in version 4.0.0. has no property `upgradeReq`.
 But BidirectionalWebsocketRouter is [using it](https://github.com/bigstepinc/jsonrpc-bidirectional/blob/master/src/BidirectionalWebsocketRouter.js#L52).

I solve id with adding request manually from `connect` event:

```javascript
wsServer.on('connection', (ws, req) => {
        ws.upgradeReq = req;
        const clientID = wsRouter.addWebSocketSync(ws);
});
```

**Comment by oxygen** [comment]
Yeah, this library has only been tested with ws 2.4~ 
Will update soon.

**Comment by oxygen** [comment]
**# Solved.**

Support for ws>=4.x is now added.
This closes #26
